### DISCUSSION TITLE
Add ability to change the background/wallpaper image

### DISCUSSION THREAD

**Discussion by LightTreasure** [labithiotis/zed-themes#86] [Ideas]
This website is a great way for users to try out themes.

I think it should offer a way to change the default background/wallpaper image:

- Especially for themes using translucency/blur, the background image greatly affects how it looks
- Users trying out themes could upload their current wallpapers and see which theme(s) would work best for them
- Theme creators can test their themes using different background images

**Comment by labithiotis** [comment]
@LightTreasure Thanks for the great suggestion, this is something we could consider. 

I see two approaches for this:
1. We do this client-side/local by converting images to data URLs (base64) and using them for background. However, refreshing the page would lose it unless we store it in local storage, but there's a five  MB storage limit, so this solution isn't the best UX.
2. We only allow custom background images for sign-in users and use Cloudflare images API, we can also do image transformations to ensure quicker load times for images.

I am likely to lean towards 2 even though it has a little more infra setup it will be best UX

**Comment by labithiotis** [comment]
@LightTreasure you can now set your own background images in preview :)
